### ISSUE TITLE
Number of stripes is non-configurable

### ISSUE THREAD

**Issue by parsonsmatt** [scrive/pool#13]
Version 2 would accept an `Int` parameter for how many stripes to create. Version 3 uses `getNumCapabilites` unconditionally.

We're running into problems upgrading, and I think this may be related. Previously, with a `createPool _ _ 1 3600 4` call, you'd have `1` stripe and `4` resources per stripe. But now you will have 4 stripes, with 1 resource per stripe. Locally, I'm seeing errors about "refusing to create resource 12", when we have a `createPool _ _ 1 3600 4` call.

`getNumCapabilities` on my machine returns `20`, so when we calculate the max resources, we get `4` (4 resources, 1 stripe). Then we get 20 stripes, each with 1 resource each.

I'd suggest adding a field for configuring stripe count on `PoolConfig` - that should alleviate this issue.

**Comment by arybczak** [comment]
> Then we get 20 stripes, each with 1 resource each.

Out of curiosity, why are you running 20 capabilities? Do you have a machine with 20 physical cores?

**Comment by parsonsmatt** [comment]
Yes, my laptop has 20 cores. `getNumCapabilities >>= print` shows `20`.

**Comment by arybczak** [comment]
I see. How many of these are physical? Asking since according to my experimentations running more capabilities than physical cores (and if you run with `+RTS -N`, `getNumCapabilities` will give you the number of logical cores) reduces performance, most likely due to GC overhead (and perhaps redundant context switches).

**Comment by parsonsmatt** [comment]
Huh apparently I have 14 physical cores, 8 "effiicency" and 6 "performance." Wild! And I guess hyperthreading gets me up to 20.
